core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -218,6 +233,15 @@ class BrokerTopicStats {
     }
   }
 
+  // This method only remove bytesIn, bytesOut, and messagesIn metrics
+  // of a broker that stops being a leader
+  def removeOldLeaderMetrics(topic: String) {
+    val topicMetrics = topicStats(topic)
+    topicMetrics.removeMetricHelper(BrokerTopicStats.MessagesInPerSec, topicMetrics.tags)
```

Comment:
To be defensive, we want to check topicMetrics is not null before using it.

Reply:
Thanks! I've added the check.